rest-api/category/models/category-models.js

```diff
@@ -0,0 +1,24 @@
+const db = require('../../../database/dbConfig')
+
+module.exports = {
+    findAllCategories,
+    findCategoryBy,
+    addCategory
+}
+
+async function findAllCategories() {
+    const categories = await db('categories');
+    return categories;
+}
+
+async function findCategoryBy(info) {
+    const category = await db('categories')
+        .where({ id: info })
+        .orWhere({ name: info });
+    return category;
+}
+
+async function addCategory(name) {
+    const [category] = await db('categories').returning('*').insert(name);
```

Comment:
Hmm. Not sure I understand quite what's going on with this line. Could you explain it to me in a call later? :)

Reply:
On this thing what he's doing is selecting the first item of the array that was returned that's why it's being deconstructed into an array. This can be made obsolete with `.returning('*')` after `.insert()`